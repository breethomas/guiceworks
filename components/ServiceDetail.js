// @flow
import React from 'react'
import css, { media2 } from '../styles/css'
import Article from './Article'
import Header from './Header'
import Markdown from './Markdown'
import PageSection from './PageSection'
import ImageIcon from './ImageIcon'
import type { ServiceProps } from '../types/app.js.flow'

const path = '/static/icons/services'

type Props = {
  service: ServiceProps
}

const headerStyle = css(
  {
    marginBottom: 80,
  },
  media2({
    marginBottom: 120,
  }),
)

export default (props: Props) => (
  <PageSection>
    <Article key={new Date()}>
      <Header className={headerStyle}>
        <ImageIcon
          alt={props.service.title}
          src={[path, props.service.icon].join('/')}
          width={70}
          height={70}
        />
      </Header>
      <Markdown
        containerProps={{ 'data-service': props.service.icon.replace('.svg', '') }}
        source={props.service.body}
      />
    </Article>
  </PageSection>
)
