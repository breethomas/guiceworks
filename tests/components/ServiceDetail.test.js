import React from 'react'
import renderer from 'react-test-renderer'
import ServiceDetail from '../../components/ServiceDetail'

describe('ServiceDetail', () => {
  let component

  beforeEach(() => {
    const service = {
      body: 'Service 1 _body_',
      icon: 'icon.svg',
      id: 'service-1',
      title: 'Service 1 title',
    }

    component = (
      <ServiceDetail
        service={service}
      />
    )
  })

  it('shows the correct snapshot for the ServiceDetail component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
