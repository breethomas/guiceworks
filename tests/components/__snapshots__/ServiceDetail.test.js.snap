// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceDetail shows the correct snapshot for the ServiceDetail component tree 1`] = `
<section
  className={
    Object {
      "data-css-24spax": "",
    }
  }
>
  <article
    className="css-1d3ulb6 css-1jyyyen"
  >
    <header
      className={
        Object {
          "data-css-hyshfv": "",
        }
      }
    >
      <img
        alt="Service 1 title"
        height={70}
        src="/static/icons/services/icon.svg"
        width={70}
      />
    </header>
    <div
      className="css-js1aix"
      data-service="icon"
    >
      <p>
        Service 1 
        <em>
          body
        </em>
      </p>
    </div>
  </article>
</section>
`;
